### PR TITLE
sqlparser: make index names optional

### PR DESCRIPTION
**Pull request by dt**
MySQL accepts index definitions without explicit names, e.g. 'UNIQUE KEY (a, b)'.

Signed-off-by: David Taylor <tinystatemachine@gmail.com>

### CODE REVIEW

File: go/vt/sqlparser/sql.y [hunk 1]
```diff
@@ -1114,26 +1115,36 @@ equal_opt:
...
+name_opt:
+  ID
+  {
+    $$ = string($1)
+  }
+| /* empty */
+  {
+    $$ = ""
+  }
+
```
Comment (sougou) [inline go/vt/sqlparser/sql.y]: Change is good.
Some nits:
* Existing code always lists the empty rule first. No particular reason, but it's better to follow existing convention.
* We generally don't mention `empty`. But there's no harm. However, let's use `//` instead.
* Also, lower level rules are listed below higher level ones, which means this should be moved after `index_or_key`.

**Comment by sougou** [comment]
Need conflicts resolved.

**Comment by sougou** [comment]
I'm surprised (and worried) that the previous merge didn't cause a conflict with sql.go on this PR.

**Comment by dt** [comment]
I rebased it on top of #4168 while I was amending it (since that one was already approved) to avoid the conflict at merge — so this briefly briefly two commit PR until that one merged, at which point it went back to being just the one, but still applying cleanly.

**Pull request merged**